### PR TITLE
PR - Code review - Space Traveller App

### PR DESCRIPTION
**Pull request by wickathou**
Space Traveller is an app that allows visitors to book rockets and missions from Spacex. The app is built using React, Redux, Bootstrap and Axios.

Features:
- Capability to book missions and rockets
- Allows users to see which rockets and missions they have booked
- Unit testing for missions and rockets
- Bootstrap for styling
- Visual indicators for loading and error states

[Live view](https://space-traveller.onrender.com/space-traveller/)

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  return (
+    <Routes>
+      <Route path="/" element={<Layout status={status} />}>
+        <Route index element={<Rockets rockets={rocketList} />} />
+        <Route path="rockets" element={<Rockets rockets={rocketList} />} />
+        <Route path="my-profile" element={<MyProfile myRockets={myRockets} />} />
+        <Route path="missions" element={<Missions />} />
+        <Route path="*" element={<NotFound />} />
+      </Route>
+    </Routes>
```
Comment (ichala) [inline src/App.js:20]: - [ ] kindly fix the root route bug, when I start the server the app opens the `http://localhost:3000/space-traveller` link as a root so it causes a 404 error , kindly fix it 🙏 


image

image

File: src/__test__/Rockets.test.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+//   //   expect(tree).toMatchSnapshot();
+//   // });
+
+//   // it('renders Calculator element correctly', () => {
+//   //   const tree = renderer
+//   //     .create(<Calculator />)
+//   //     .toJSON();
+//   //   expect(tree).toMatchSnapshot();
+//   // });
+// });
```
Comment (ichala) [inline src/__test__/Rockets.test.js]: - [x] to keep you repo clean and professional kindly remove any commented code from ur project 🙏

File: src/components/MissionTable/MissionTable.js [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+  const joinMissionHandler = (e) => {
+    const { target: { id } } = e;
+    dispatch(joinMission(id));
+  };
+
+  const leaveMissionHandler = (e) => {
+    const { target: { id } } = e;
+    dispatch(leaveMission(id));
+  };
+
```
Comment (ichala) [inline src/components/MissionTable/MissionTable.js]: - [ ] Please note that the mission state should only be updated once in the first visit, currently it fetches the data & resets the state every time I change routes, kindly fix it 🙏 


statebug

File: src/__test__/Missions.test.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+    const missions = render(
+      <Provider store={store}>
+        <BrowserRouter>
+          <Missions />
+        </BrowserRouter>
+      </Provider>,
+    );
+    expect(missions).toMatchSnapshot();
+  });
+});
```
Comment (ichala) [inline src/__test__/Missions.test.js]: - [ ] as part of the requirements you should  **_add tests for all your components and functions._** currently you tested only two components , kindly tests all of your components and add unit tests for your functions 🙏

**Review by ichala — CHANGES_REQUESTED** [review]
Hi Team
Good job so far! 
👏👏👏 
You still need to work on some issues to go to the next project, but you're almost there!

the-simpsons-homer-simpson

## To Highlight 🎉

- No Linter errors  ✔️ 
- Good Design ✔️


 ## Required Changes ♻️

- [ ] **_🛑  Check The Comments Below_**





## Optional suggestions

- N/A

**********
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/MissionTable/MissionTable.js (hunk 3 above)
Comment (wickathou) [reply src/components/MissionTable/MissionTable.js]: @Talha-Hanif5666

File: src/__test__/Missions.test.js (hunk 4 above)
Comment (wickathou) [reply src/__test__/Missions.test.js]: - Done for components Rockets, MyRockets, Layout and Navbar
- Done for all extra routes

@Talha-Hanif5666 Missions is pending

**Review by ghost — APPROVED** [review]
Hi Team 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights

✅ Correct use of Gitflow
✅ No linter error
✅ App works as expected 👍 
✅ All tests are passing 👍 

## Optional suggestions

N/A

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**